src/brsTypes/components/RoSGNode.ts

```diff
@@ -767,3 +768,25 @@ function addFields(interpreter: Interpreter, node: RoSGNode, typeDef: ComponentD
         }
     }
 }
+
+function addChildren(interpreter: Interpreter, node: RoSGNode, typeDef: ComponentDefinition) {
+    let children = typeDef.children;
+    let appendChild = node.getMethod("appendchild");
+
+    children.forEach(child => {
+        let newChild = createNodeByType(interpreter, new BrsString(child.name));
+        if (newChild instanceof RoSGNode && appendChild) {
+            appendChild.call(interpreter, newChild);
+            let setField = newChild.getMethod("setfield");
+            for (let [key, value] of Object.entries(child.fields)) {
+                if (setField) {
+                    setField.call(interpreter, new BrsString(key), new BrsString(value));
```

Comment:
Do we know that all of the fields will be `BrsString`?

Reply:
The only other type I've seen in these fields is an array type (i.e. `translation="[1,20]"`) but there is no way to know the type unless all built-in Roku components are loaded. I think for now this should be fine until we know what we'll do with the built-in nodes